src/libraries/System.Text.Json/gen/Helpers/EnumFormatter.cs

```diff
@@ -0,0 +1,109 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+namespace System.Text.Json.SourceGeneration
+{
+    public static class EnumFormatter<TEnum> where TEnum : struct, Enum
```

Comment:
This is a ton of code duplicating logic in Enum itself.  Can you instead just do something like:
```C#
static string FormatEnum<TEnum>(TEnum value) where TEnum : Enum
{
    string[] values = value.ToString().Split(',', StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
    return string.Join(" | ", values.Select(s =>
        char.IsAsciiDigit(s[0]) ?
            $"({typeof(TEnum).Name}){s}" :
            $"{typeof(TEnum).Name}.{s}"));
}
```
?

Reply:
I could, but I figured doing that was preferable to having to parse the `.ToString()` output for fearing of missing a corner case.

Reply:
The code here also misses corner cases.  For example, this:
```C#
Console.WriteLine(FormatEnum(AttributeTargets.All));
Console.WriteLine();
Console.WriteLine(EnumFormatter<AttributeTargets>.FormatEnumLiteral(nameof(AttributeTargets), AttributeTargets.All));
```
produces this:
```
AttributeTargets.All

AttributeTargets.Assembly | AttributeTargets.Module | AttributeTargets.Class | AttributeTargets.Struct | AttributeTargets.Enum | AttributeTargets.Constructor | AttributeTargets.Method | AttributeTargets.Property | AttributeTargets.Field | AttributeTargets.Event | AttributeTargets.Interface | AttributeTargets.Parameter | AttributeTargets.Delegate | AttributeTargets.ReturnValue | AttributeTargets.GenericParameter
```
I'd rather have the formatting be done by the tried-and-true implementation than try to handle all the various cases the Enum implementation already handles.  We know what Enum.ToString can produce, and we can have tests for various cases.  If we encounter something really remote that we somehow missed, we can fix it.

My real hesitancy is that this is both a lot of logic while still being incomplete.

Reply:
I've updated the PR using a variation of your proposed solution.